src/app/reactive-templates/reactive-components/row-base.component.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Component,
+  computed,
+  HostBinding,
+  inject,
+  InjectionToken,
+  input,
+  OnDestroy,
+  OnInit,
+  signal,
+  Signal,
+} from "@angular/core";
+import { FlowTypes } from "src/app/shared/model";
+import { VariableStore } from "../stores/variable-store";
+import { RowService } from "../services/row.service";
+import { Parameters } from "./parameters";
+import { NamespaceService } from "../services/namespace.service";
+import { ActionService } from "../services/action.service";
+
+export const ROW_PARAMETERS = new InjectionToken<Parameters>("ROW_PARAMETERS");
+
+@Component({
+  selector: "oab-row-base",
+  template: ``, // template is empty, to be overridden by child components
+})
+export abstract class RowBaseComponent<TParams extends Parameters> implements OnInit, OnDestroy {
+  public row = input.required<FlowTypes.TemplateRow>();
+  public namespace = input("");
+  public name = computed(() =>
+    this.namespaceService.getFullName(this.namespace(), this.row().name)
+  );
+  public value: Signal<any>;
+  public condition = signal(true);
+  public params = inject(ROW_PARAMETERS) as TParams;
+
+  private dependantVariables: string[] = [];
+
+  protected variableStore = inject(VariableStore);
+  protected rowService = inject(RowService);
+  protected namespaceService = inject(NamespaceService);
+  protected actionService = inject(ActionService);
+
+  protected subscriptions = [];
+
+  @HostBinding("style.display")
+  get displayStyle() {
+    return this.condition() ? "" : "none";
+  }
+
+  /**
+   * All RowBaseComponent implementations should call this method (`super.ngOnInit();`) in their ngOnInit lifecycle hook.
+   */
+  ngOnInit(): void {
+    const row = this.row();
+
+    this.value = this.variableStore.asSignal(this.name());
+    this.condition.set(this.rowService.evaluateCondition(row, this.namespace()));
+
+    this.setParams();
+    this.setDependencies();
+    this.watchDependencies();
+
+    // Set default value
+    this.variableStore.set(this.name(), this.rowService.evaluateValue(row, this.namespace()));
+  }
+
+  /*
+   * Sets the rows value and updates the variable store.
+   */
+  public setValue(value: any): void {
+    this.row().value = value;
```

Comment:
Is this the proper way to update a nested value on a signal – i.e. does mutating the properties of `row()` trigger signal updates for any dependencies of `row()`? I'm not sure. Appreciate that we can't update `row()` directly since it's an input signal. But can this line be removed entirely, since we're managing the state via the variable store?